### PR TITLE
[onert-micro] This PR adds L2Normalization kernel

### PR DESCRIPTION
**Pull request by Torrero**
This commit adds L2Normalization kernel

Kernels support. Support for the L2Normalization kernel for the [issue](https://github.com/Samsung/ONE/issues/12427)

ONE-DCO-1.0-Signed-off-by:  Evgenii Maltsev e.maltsev@samsung.com

### CODE REVIEW

File: onert-micro/onert-micro/src/import/kernels/L2Normalize.cpp [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  OMRuntimeShape input_shape(input);
+  OMRuntimeShape output_shape(output);
+
+  status = utils::checkCondition(input_shape.flatSize() == output_shape.flatSize());
+  if (status != Ok)
+    return status;
+
+  status = utils::checkCondition(input_shape.dimensionsCount() == output_shape.dimensionsCount());
+  if (status != Ok)
+    return status;
```
Comment (SlavikMIPT) [inline onert-micro/onert-micro/src/import/kernels/L2Normalize.cpp:66]:
```suggestion
  status = utils::checkCondition(input_shape == output_shape);
```

Comment (Torrero) [reply onert-micro/onert-micro/src/import/kernels/L2Normalize.cpp]: Changed it. Thank you.

File: onert-micro/onert-micro/src/execute/kernels/L2Normalize.cpp [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+      const auto dim_count = inputs_shape.dimensionsCount();
+
+      const auto trailing_dim = dim_count - 1;
+
+      int flat_size = 1;
+      for (int i = 0; i < inputs_shape.dimensionsCount(); ++i)
+      {
+        flat_size *= (i == trailing_dim) ? 1 : inputs_shape.dims(i);
+      }
```
Comment (BalyshevArtem) [inline onert-micro/onert-micro/src/execute/kernels/L2Normalize.cpp:90]: PALUtils (in pal/common) contains helper function - `flatSizeSkipDim` . I think you can pass the input_shape into pal::L2Normalization and use there `flatSizeSkipDim`

File: onert-micro/onert-micro/include/pal/common/PALL2Normalize.h [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+  }
+
+  return Ok;
+}
+
+} // namespace pal
+} // namespace execute
+} // namespace onert_micro
+
+#endif // ONERT_MICRO_EXECUTE_PAL_LOG_SOFTMAX_COMMON_H
```
[the remark this answers is not in the document]
Comment (Torrero) [reply onert-micro/onert-micro/include/pal/common/PALL2Normalize.h]: fixed it, thank you

File: onert-micro/onert-micro/src/execute/kernels/L2Normalize.cpp (hunk 2 above)
Comment (Torrero) [reply onert-micro/onert-micro/src/execute/kernels/L2Normalize.cpp]: used `flatSizeSkipDim` here, thank you

**Review by BalyshevArtem — COMMENTED** [review]
LGTM

File: onert-micro/onert-micro/src/import/kernels/L2Normalize.cpp [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+  status = utils::checkCondition(input->type() == output->type());
+  if (status != Ok)
+    return status;
+
+  OMRuntimeShape input_shape(input);
+  OMRuntimeShape output_shape(output);
+
+  status = utils::checkCondition(input_shape == output_shape);
+
+  return status;
```
Comment (BalyshevArtem) [inline onert-micro/onert-micro/src/import/kernels/L2Normalize.cpp:62]: ditto

**Review by BalyshevArtem — APPROVED** [review]
LGTM

**Pull request merged**